core/services/workflows/workflow_spec_factory.go

```diff
@@ -0,0 +1,51 @@
+package workflows
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+var ErrInvalidWorkflowType = errors.New("invalid workflow type")
+
+type SDKWorkflowSpecFactory interface {
+	GetSpec(rawSpec, config []byte) (sdk.WorkflowSpec, error)
+	GetRawSpec(wf string) ([]byte, error)
+}
+
+type WorkflowSpecFactory map[job.WorkflowSpecType]SDKWorkflowSpecFactory
+
+func (wsf WorkflowSpecFactory) ToSpec(workflow string, config []byte, tpe job.WorkflowSpecType) (sdk.WorkflowSpec, string, error) {
+	if tpe == "" {
+		tpe = job.DefaultSpecType
+	}
+
+	factory, ok := wsf[tpe]
+	if !ok {
+		return sdk.WorkflowSpec{}, "", ErrInvalidWorkflowType
+	}
+
+	rawSpec, err := factory.GetRawSpec(workflow)
+	if err != nil {
+		return sdk.WorkflowSpec{}, "", err
+	}
+
+	spec, err := factory.GetSpec(rawSpec, config)
```

Comment:
@nolag Will we ever want to call GetSpec + GetRawSpec separately? If not I think it makes sense to combine the two functions into one.